internal/buildpackage/builder.go

```diff
@@ -112,25 +112,27 @@ func (p *PackageBuilder) Save(repoName string, publish bool) (imgutil.Image, err
 }
 
 func validateBuildpacks(mainBP dist.Buildpack, depBPs []dist.Buildpack) error {
-	depsWithRefs := map[dist.BuildpackInfo][]dist.BuildpackInfo{}
+	depsWithRefs := map[string][]dist.BuildpackInfo{}
 
 	for _, bp := range depBPs {
-		depsWithRefs[bp.Descriptor().Info] = nil
+		depsWithRefs[bp.Descriptor().Info.FullName()] = nil
```

Comment:
Don't use the full struct because `Info.Homepage` might not have been populated.